Add first version of the a11y guide

jgiardino opened the pull request:
This adds the first version of the a11y guide.

Link: https://deploy-preview-196--pf-next.netlify.com/accessibility-guide

Note: additional css is needed to handle how .md files present lists and nested lists.

patternfly-build commented:
Deploy preview for *pf-next* ready!

Built with commit fc1c2e6106b004956f7d700502876156e6db4efa

https://deploy-preview-196--pf-next.netlify.com

jgiardino commented:
Thanks for the review, @michael-coker! I addressed most of your suggestions. 

> "as well as physical, sensory or learning" - needs an oxford comma

For this one, I left the oxford comma out since this sentence is already comma heavy.

I'm still on the fence regarding the best way to structure items in the **Design and Development Guidelines** section, so if anyone has suggestions, please let me know. I will probably make refinements in future iterations too.

andresgalante commented:
@jgiardino I think this is great for alpha 1, and we'll probably improve the Design and Development Guidelines section as problems appear on PRs.

there is a conflict on `src/site/workspace.scss` would you like me to help rebasing your branch?

jgiardino commented:
Thanks, @andresgalante! I just rebased. Let me know if any changes are needed.

matthewcarleton commented on src/site/pages/accessibility-guide.md:
```diff
@@ -0,0 +1,86 @@
+      - [What PatternFly Designers and Developers Should Address](#what-patternfly-designers-and-developers-should-address)
+      - [What Product Developers and Designers Should Address](#what-product-developers-and-designers-should-address)
+  - [Testing](#testing)
+  - [Screen reader support](#screen-reader-support)
```
should this be "Screen Reader Support"?

matthewcarleton commented:
looks great Jenn!

srambach commented on src/site/pages/accessibility-guide.md:
```diff
@@ -0,0 +1,86 @@
+
+### Low Vision
+
+Users with low vision can have different needs depending on their eye condition. Needs can range from not relying on color to communicate information, color palettes that provide sufficient contrast, and a layout that is responsive when the user zooms in to magnify the contents of the screen.
```
"Eye condition" sounds a little odd to me. How about "depending on the nature of their visual impairment." ? Also, the next sentence is complicated - how about something like "Users may have difficulty with color differentiation, blurriness, or lack of vision in central or peripheral areas. These needs mean that interfaces should not rely on color to communicate information, palettes need to have sufficient contrast, and screens should behave well when a screen magnifier is used."

srambach commented:
Looks great, @jgiardino! Thank you for all the work you've put into it, especially highlighting and linking the key parts of the spec.
(However, I stand with @michael-coker on the need for the Oxford comma)
:-)

stacymcauliffe commented on src/site/pages/accessibility-guide.md:
```diff
@@ -0,0 +1,86 @@
+Some users fall into just one category above, but it’s possible for a user to fall into more than one, or use tools and devices designed for a user group they don’t fall into. One of the greatest benefits of inclusive design practices is that methods that are designed for a specific user group will often provide benefits to all users.
+
+
+## Checklists
```
We might try something like this --
Get rid of the Checklists title and make the sections as follows:
-Design and Development Guidelines 
-Accessibility Checklist for PatternFly Designers and Developers
-Accessibility Checklist for All Product Designers and Developers

jgiardino replied:
I've definitely struggled with how to label this second main section. I'd like to think about it more before making any changes now.

jgiardino commented on src/site/pages/accessibility-guide.md:
```diff
@@ -0,0 +1,86 @@
+*Please note, this guide is a work in progress and will be updated regularly. We welcome your comments and feedback.*
+
+The goal of software accessibility is to remove barriers to the use of technology so that it can be inclusive of everyone, regardless of physical ability. Accessibility addresses permanent to temporary disabilities and includes a range of visual impairments, as well as physical, sensory or learning challenges, including those that occur normally as adults age. Designing Red Hat software with this ...
+
```
solves the oxford comma dilemma ;-)

jgiardino commented on src/site/pages/accessibility-guide.md:
```diff
@@ -0,0 +1,86 @@
+
+### No Vision
+
+Users with no vision rely on screen readers to access web sites and applications. Often, screen reader users will navigate a page by browsing specific elements on the page, like headers, links, or form elements. Using semantic elements and ensuring that these elements are still meaningful when pulled out of context is critical for the usability of the site.
```
How about this for the last statement instead: "Use semantic elements and check that labels are meaningful when pulled out of context."

These are really two separate action items to help facilitate methods used by screen reader users.

stacymcauliffe replied:
Yeah, this sounds good to me!

srambach commented:
That change sounds good, @jgiardino

stacymcauliffe commented:
This looks good to me, Jenn! I dont have any more suggestions.

dgutride commented:
Addresses https://github.com/patternfly/patternfly-next/issues/107
